### ISSUE TITLE
Ciudad Mexico

### ISSUE THREAD

**Issue by mpadge** [ropensci/bikedata#55]
@RobinLovelace Can you please provide details of the Mexico City data you mentioned? It'd be great to incorporate that if possible

**Comment by Robinlovelace** [comment]
Sure:

```r
# from: https://www.ecobici.cdmx.gob.mx/es/informacion-del-servicio/open-data
u = "https://www.ecobici.cdmx.gob.mx/sites/default/files/data/usages/2018-01.csv"
u_static = "https://www.ecobici.cdmx.gob.mx/sites/default/files/data/usages/"
download.file(url = u, destfile = "input-data/month-data.csv")

# create loop to download every month
yrs = 2011:2018
months = formatC(1:12, width = 2, flag = "0")
for(y in yrs) {
  for(m in months) {
    um = paste0(u_static, y, "-", m, ".csv")
    print(um)
    
    destfile = paste0("input-data/", y, "-", m, ".csv")
    
    download.file(url = um, destfile = destfile)
    
  }
}
```

**Comment by Robinlovelace** [comment]
This shows a first attempt at linking them with OSM data. The results are unreliable because the docking stations do not have correct names. 

```r
# from: https://www.ecobici.cdmx.gob.mx/es/informacion-del-servicio/open-data
u = "https://www.ecobici.cdmx.gob.mx/sites/default/files/data/usages/2018-01.csv"
u_static = "https://www.ecobici.cdmx.gob.mx/sites/default/files/data/usages/"
download.file(url = u, destfile = "input-data/month-data.csv")

# create loop to download every month
yrs = 2011:2018
months = formatC(1:12, width = 2, flag = "0")
for(y in yrs) {
  for(m in months) {
    um = paste0(u_static, y, "-", m, ".csv")
    print(um)
    
    destfile = paste0("input-data/", y, "-", m, ".csv")
    
    download.file(url = um, destfile = destfile)
    
  }
}
```

**Comment by Robinlovelace** [comment]
Thanks to @luis-tona for getting me onto this, he's on it!

Any updates Luis? This may lead the bikeshare data to be made available online for everyone (which could be an amazing side benefit of the project).

**Comment by mpadge** [comment]
Great, I'll check out the data source asap. A huge part of˙bikedata` is just data cleaning and stuff like station name matching, so no biggy there. I'll just have to ensure an ongoing commitment to data availability

**Comment by mpadge** [comment]
Data sources can be extracted as "https://www.ecobici.cdmx.gob.mx/sites/default/files/data/usages/YYYY-MM.csv", starting at 2010-02.

**Comment by mpadge** [comment]
The trip files have only station numbers. These are all given on [the official station map](https://www.ecobici.cdmx.gob.mx/en/stations-map), but the raw data are all buried in a drupal scheme, so can't be accessed. There's supposed to be an API, but I applied for a key and just received an empty email. I've written to ask them about providing data, and/or help with the API, and will have to wait for a response.

**Comment by Robinlovelace** [comment]
Tricky. I know Luis is struggling to make sense of the data also which must be frustrating as it seems to all be there, just not in a joined-up way. Let us know how you get on + cheers for the updates.

**Comment by mpadge** [comment]
code to get the data files:
```
data_dir <- tempdir ()
url_base <- "https://www.ecobici.cdmx.gob.mx/sites/default/files/data/usages/"
furl <- paste0 (url_base, "2010-02.csv")
furl <- paste0 (url_base, "2017-02.csv")
destfile <- file.path (data_dir, paste0 ("mx", basename(furl)))
resp <- httr::GET (furl, httr::write_disk (destfile, overwrite = TRUE))
````

**Comment by Robinlovelace** [comment]
Great progress - is that all of them though? Heads-up @luis-tona - may be of use for your project.

**Comment by mpadge** [comment]
nah, that's just me dumping the generic code. Any `"YYYY-MM.csv"` can be entered. The annoying problem remains that the stations can't be accessed, and I've not got any reply to my query to ecobici. In the absence of a response, the whole incorporation will have to be shelved until station data become available

**Comment by mpadge** [comment]
Note that [GBFS data](https://github.com/NABSA/gbfs/blob/master/systems.csv) are available for Guadlajara, but not Cuidad Mexico. Still have to wait on this one.

**Comment by Robinlovelace** [comment]
We've made great progress cleaning the datasets - 100 million+ journeys (rows of data) logged.

**Comment by Robinlovelace** [comment]
We've made great progress cleaning the datasets - 100 million+ journeys (rows of data) logged.
